tests/components/derivative/test_sensor.py

```diff
@@ -354,6 +355,31 @@ async def test_suffix(hass: HomeAssistant) -> None:
     assert round(float(state.state), config["sensor"]["round"]) == 0.0
 
 
+async def test_total_increasing_reset(hass: HomeAssistant) -> None:
+    """Test derivative sensor state with total_increasing sensor input."""
+    # When the input sensor is reset, the derivative sensor should not record a negative change.
+    times = [20, 30, 40]
+    values = [10, 30, 0]
+    config, entity_id = await _setup_sensor(hass, {"unit_time": UnitOfTime.SECONDS})
+
+    base = dt_util.utcnow()
+    with freeze_time(base) as freezer:
+        for time, value in zip(times, values, strict=False):
+            freezer.move_to(base + timedelta(seconds=time))
+            hass.states.async_set(
+                entity_id,
+                value,
+                {ATTR_STATE_CLASS: SensorStateClass.TOTAL_INCREASING},
+                force_update=True,
+            )
+            await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.power")
+    assert state is not None
+
+    assert round(float(state.state), config["sensor"]["round"]) == 2.00
```

Comment:
Instead of just the magic number `2.00`, calculate it or add a comment explaining it

Reply:
at the second thought...
* I don't want to calculate it because it would duplicate the implementation in the unit tests. I want the tests to be calculation-free as much as possible so that they're independent of the algorithm.
* There's no need to comment because the test is already clear. My existing comment isn't unique to this unit test but applies to the entire component. The existing web documentation pointing to Wikipedia should suffice.

Therefore, I will remove this comment in the next commit. Wdyt?